### ISSUE TITLE
[Feature Request] Double-clicking and the Escape Key to open and close full-screen forms

### ISSUE THREAD

**Issue by acoard-aot** [TIBCOSoftware/flogo#545]
This is really two related but separate feature requests.  I'll number them separately within each answer.  Both of these are related to the Flow view.

Feature 1: Pressing escape to close forms.
Feature 2: Double-clicking to open forms.

**Current behavior:** 

1. When the user has the Triggers or Activities forms open ("fullscreen forms"), pressing the scape key does nothing.  The user can only close the form via clicking the tiny 'x' icon in the top right.
2. In the Flow view, if the user double-clicks on a Trigger or Activity, nothing happens.  In order to open the details a user must click the "..." icon, then the "gear" icon.

**Expected behavior:**

1. When the user has the Triggers or Activities forms open ("fullscreen forms"), pressing the scape key should close the form.
2. If the user double-clicks on a Trigger or Activity, it should be as if the gear icon was clicked.

**What is the motivation / use case for changing the behavior?**

Power-users that use the  GUI  have to open and close these forms a ton, probably  dozens or hundreds (or thousands) of times.  Adding these features would benefit power users and not  negatively impact more casual users.  In  fact, the UX here is familiar and should work for casual users too.  

**Additional information you deem important (e.g. I need this tomorrow):**

I would really  value the 'Escape' key.  I understand if you don't like double-clicking, as that's more desktop than web, *but* escape key to close modal forms is common in the web too.  Thanks.

**Comment by fcastill** [comment]
I agree, we need to enhance keyboard accessibility in the Web UI. I've opened a Github issue in the Web UI repo to reference this one: https://github.com/project-flogo/flogo-web/issues/1333 And I think we can fix the `esc` issue soon. Please let us know if there are other issues related to keyboard navigation that you consider particularly important.

Regarding the double click behavior on a tile, I like what you're proposing because it is kind of like opening an application in an OS desktop. However, just like you mentioned, double-clicking is not common in web apps. I feel that the issue is that the 'gear' icon is hidden away in an overflow menu when it should be treated like a primary action. I think we should update the UI to make the configuration modal easier to reach instead of providing an alternate method to open the configuration modal (like a double click). Do you think that would fulfill your request?
